src/components/ObsDetails/FlagItemModal.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+
+import CheckBox from "@react-native-community/checkbox";
+import createFlag from "api/flags";
+import Button from "components/SharedComponents/Buttons/Button";
+import {
+  Modal,
+  SafeAreaView,
+  Text, View
+} from "components/styledComponents";
+import { t } from "i18next";
+import type { Node } from "react";
+import React, { useRef, useState } from "react";
+import {
+  findNodeHandle
+} from "react-native";
+import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
+import { TextInput } from "react-native-paper";
+import IconMaterial from "react-native-vector-icons/MaterialIcons";
+import useAuthenticatedMutation from "sharedHooks/useAuthenticatedMutation";
+
+type Props = {
+  id:number,
+  showFlagItemModal: boolean,
+  closeFlagItemModal: Function
+}
+
+const FlagItemModal = ( {
+  id, showFlagItemModal, closeFlagItemModal
+}: Props ): Node => {
+  const keyboardScrollRef = useRef( null );
+  const [value, setValue] = useState( "none" );
+  const [explanation, setExplanation] = useState( "" );
+
+  const scrollToInput = node => {
+    keyboardScrollRef?.current?.scrollToFocusedInput( node );
+  };
+
+  const toggleValue = checkbox => {
+    if ( value === checkbox ) { setValue( "none" ); } else { setValue( checkbox ); }
+  };
+
+  const clearForm = () => {
```

Comment:
This isn't really a form in the HTML sense, so I think a description like `resetFlagModal` might make more sense here. What do you think?

Reply:
I treated the checkboxes as a form with radio buttons but renaming to `resetFlagModal` could make sense as well. I don't have a strong preference so i'll change it to `resetFlagModal` for now.